### ISSUE TITLE
Minimal case should do more

### ISSUE THREAD

**Issue by scarabeusiv** [openSUSE/spec-cleaner#76]
In minimal mode we just remove whitespace and curlify some generic stuff.

But we should do all the trivial changes that are not fatal like updating license or removing %clean section.

**Comment by scarabeusiv** [comment]
@coolo @DimStar77 @lnussel guys could you provide me some insights of what is absolute must for the minimal mode (eg what functionality must be 100% covered from the format_spec_file)

**Comment by lnussel** [comment]
I don't even know what prepare_spec does from the top of my head. We'd need a list of thing to decide :-)

**Comment by coolo** [comment]
prepare_spec does (I'm sure I forget things, but here we go):

- [ ] Rewrite the license according to the mapping

- [ ] Indent the header to have the : at the same position if possible

- [ ] Add a license plate at the beginning preserving non-suse copyrights

- [ ] reorder headers to be one require, buildrequire, provide in one line and sorted

- [ ] remove %changelog content

- [ ] capitalize docdir, arch, patch, source, req and prov

- [ ] add a new line before each section

- [ ] remove needsrootforbuild, needsbinariesforbuild, %debug_package and Epoch

- [ ] remove Autoreqprov: (on|yes)

- [ ] replace BuildArchitectures with BuildArch

- [ ] remove license from subpackages if it's the same for all

- [ ] put the license in all subpackages if there are more than one license in the spec file

**Comment by scarabeusiv** [comment]
WIP in minimal branch, remaining is to fix up the Preamble section in rpmpreamble.

**Comment by scarabeusiv** [comment]
@coolo Technically speaking we are covering now all apart from the last two points. But there is problem.

The header parsing is now done for the whole header and it gets reformated -> it produces quite huge diff for the first run on the preamble. Is that okay or should try to force just read it line-by-line and not try to be smart?

**Comment by coolo** [comment]
can you point out an example?

**Comment by scarabeusiv** [comment]
https://github.com/openSUSE/spec-cleaner/blob/minimal/tests/in/interestingheader.spec
https://github.com/openSUSE/spec-cleaner/blob/minimal/tests/out-minimal/interestingheader.spec

https://github.com/openSUSE/spec-cleaner/blob/minimal/tests/in/conditions.spec
https://github.com/openSUSE/spec-cleaner/blob/minimal/tests/out-minimal/conditions.spec

**Comment by coolo** [comment]
hmm, it would be fine for me - but I guess you would need to run this on more real life cases from O:F and provide some diffs and show it to the packaging crew. I can think of many who will have oppinions ;)

**Comment by scarabeusiv** [comment]
Ok, i will let Tcech review the merge of the branch. Create new release and ask on -packaging/-factory for testing and bugreports/complains :)

**Comment by coolo** [comment]
and I recommend to go one step further and provide a list of patches that would inherit the change.

**Comment by scarabeusiv** [comment]
List of patches? Probably don't understand what you mean by that, PR is https://github.com/openSUSE/spec-cleaner/pull/79

**Comment by coolo** [comment]
"Dear packagers mailing list, I have a proposal to obsolete format_spec_file with this great neat tool here. I reduced the impact on the spec files as much as I could, but still I recommend to go one step further in how we format the spec file, especially the preamble. Here is a list of example diffs what it would look like if you ran osc ci if using my code. I picked gcc, xteddy and postgresql94 so you have an idea - but feel free to run COMMAND A on your own spec file and give me feedback. I really think format_spec should die soon"

does that work for you? :)
